From 53c923a037b805e4e8677759819c5fbe0d0fd3cc Mon Sep 17 00:00:00 2001
From: Razvan Cojocaru <rcojocaru@bitdefender.com>
Date: Tue, 9 May 2017 12:00:17 +0300
Subject: [PATCH] x86/vm_event: don't send out CR4 PGE events if
 opt_introspection_extn

Introspection is not interested in PGE-only CR4 modifications.

Signed-off-by: Razvan Cojocaru <rcojocaru@bitdefender.com>
diff --git a/xen/arch/x86/hvm/hvm.c b/xen/arch/x86/hvm/hvm.c
index 41781bb..fa8a636 100644
--- a/xen/arch/x86/hvm/hvm.c
+++ b/xen/arch/x86/hvm/hvm.c
@@ -2426,7 +2426,8 @@ int hvm_set_cr4(unsigned long value, bool_t may_defer)
     }
 
     if ( may_defer && unlikely(v->domain->arch.monitor.write_ctrlreg_enabled &
-                               monitor_ctrlreg_bitmask(VM_EVENT_X86_CR4)) )
+                               monitor_ctrlreg_bitmask(VM_EVENT_X86_CR4)) &&
+         (!opt_introspection_extn || (old_cr ^ value) != X86_CR4_PGE) )
     {
         ASSERT(v->arch.vm_event);
 
